src/QMCWaveFunctions/EinsplineSet.cpp

```diff
@@ -1864,7 +1864,7 @@ EinsplineSetExtended<StorageType>::evaluate_notranspose(const ParticleSet& P, in
       for(unsigned a0(0); a0<OHMMS_DIM; a0++)
       {
         double olp= std::abs((PrimLattice.R[a0]*kPoints[j][a0])/M_PI +1e-6);
-        olp -= (int)olp;
+        olp -= (int)std::round(olp);
```

Comment:
This part of code seems suspicious to me. PrimLattice.R is a tensor 3x3. The loop only uses the first 3 elements.
For the checking, I would prefer to remove 1e-6 and abs in the first line and change the second line to
`olp = std:abs(olp-std:round(olp));

Reply:
@ye-luo I think you are right to suspect this loop. It tries to determine if the desired orbital `j` can be obtained from a special k-point in the primitive cell and sets the `trs` flag. The `trs` flag, in turn, determines how the derivatives are assigned. The current implementation passed all the tests I provided, but I see now that it will fail in general.

How is k-point classification done in the current spline implementation? Is there a routine I can steal instead of this somewhat ad hoc for loop?

Reply:
trs=true seems like the very very old code path to UseRealOrbitals (R2R) case which you don't need to use.
The contents should have been already in
```
EinsplineSetExtended<double>::evaluate_notranspose(const ParticleSet& P, int first, int last,
    RealValueMatrix_t& psi, RealGradMatrix_t& dpsi,
    RealHessMatrix_t& grad_grad_psi,
    RealGGGMatrix_t& grad_grad_grad_logdet)
```
So just remove the trs=true code path.